pkg/serviceaccount/jwt.go

```diff
@@ -131,7 +125,8 @@ type jwtTokenAuthenticator struct {
 }
 
 type Validator interface {
-	Validate(tokenData string, public *jwt.Claims, private *legacyPrivateClaims) error
+	Validate(tokenData string, public *jwt.Claims, private interface{}) (user.Info, error)
```

Comment:
not sure I'd move user info construction into validate... maybe have it return the service account info components instead? (name, namespace, uid). We'd want to make sure user info got built consistently